### PR TITLE
feat: add ubuntu welcome snap

### PR DESCRIPTION
**Pull request by d-loose**
Ref #2120 
UDENG-744

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -93,10 +93,24 @@ jobs:
         branch: create-pull-request/l10n
         delete-branch: true
 
-  snap:
+  ubuntu-dektop-installer-snap:
```
Comment (jpnurmi) [inline .github/workflows/ci.yml:96]: could we move this to `installer.yaml` where the installer specific jobs are, and add `welcome.yaml` for welcome-specific jobs?

File: packages/ubuntu_welcome/snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    extensions: [gnome]
+    command-chain: [bin/launcher]
+    #desktop: bin/data/flutter_assets/assets/ubuntu-welcome.desktop
+    environment:
+      PATH: $SNAP/usr/bin:$SNAP/bin:$PATH
+      LIBGL_DRIVERS_PATH: $SNAP/usr/lib/$CRAFT_ARCH_TRIPLET/dri
+      GIO_MODULE_DIR: $SNAP/usr/lib/$CRAFT_ARCH_TRIPLET/gio/modules
+      LIVE_RUN: 1
+      LOG_LEVEL: debug
+      SNAP_PYTHON: python3
```
Comment (jpnurmi) [inline packages/ubuntu_welcome/snap/snapcraft.yaml:24]: i wonder if we could leave out everything else except `LOG_LEVEL` might be nice to keep

- `LIVE_RUN` and `SNAP_PYTHON` are for subiquity
- `PATH`, `LIBGL_DRIVERS_PATH` and `GIO_MODULE_DIR` might not be needed for a non-classic snap?

File: packages/ubuntu_welcome/snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  Ubuntu Welcome description.
+grade: stable
+confinement: devmode
+base: core22
+issues: https://github.com/canonical/ubuntu-desktop-installer/issues
+contact: https://github.com/canonical/ubuntu-desktop-installer/issues
+
+apps:
+  ubuntu-welcome:
+    command: bin/ubuntu_welcome
```
Comment (jpnurmi) [inline packages/ubuntu_welcome/snap/snapcraft.yaml:14]: unrelated side-note: yikes, we should use a hyphen in the executable :)

File: packages/ubuntu_welcome/snap/local/launcher [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```
Comment (jpnurmi) [inline packages/ubuntu_welcome/snap/local/launcher:1]: do we need these hacks with a non-classic snap? would the gnome extension make sure gtk/glib and friends are correctly setup?

File: packages/ubuntu_welcome/snap/snapcraft.yaml (hunk 2 above)
Comment (d-loose) [reply packages/ubuntu_welcome/snap/snapcraft.yaml]: Removed

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -93,10 +93,13 @@ jobs:
         branch: create-pull-request/l10n
         delete-branch: true
 
-  snap:
+  ubuntu-welcome-snap:
```
Comment (jpnurmi) [inline .github/workflows/ci.yml:96]: we can remove this

File: .github/workflows/installer.yaml [hunk 6]
```diff
@@ -61,3 +61,16 @@ jobs:
         working-directory: packages/ubuntu_desktop_installer
         env:
           SUBIQUITY_REPLAY_TIMESCALE: 100
+
+  snap:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - run: ln -s packages/ubuntu_desktop_installer/snap
```
Comment (jpnurmi) [inline .github/workflows/installer.yaml]: not needed anymore?

File: .github/workflows/welcome.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  FLUTTER_VERSION: '3.10.x'
+
+jobs:
+  snap:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - run: |
+        rm snap
+        ln -s packages/ubuntu_desktop_installer/snap
```
Comment (jpnurmi) [inline .github/workflows/welcome.yaml:20]: nit-pick: `ln -sf` wouldn't need a separate `rm`?

File: .github/workflows/welcome.yaml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  workflow_dispatch:
+
+env:
+  FLUTTER_VERSION: '3.10.x'
+
+jobs:
+  snap:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
```
Comment (jpnurmi) [inline .github/workflows/welcome.yaml:17]: with: submodules: recursive

**Review by jpnurmi — APPROVED** [review]
i'm not a snap expert but this LVGTM :+1:

**Pull request merged**